grafana-plugin/src/containers/ServiceNowConfigDrawer/CompleteServiceNowConfigModal.tsx

```diff
@@ -0,0 +1,134 @@
+import React, { useState } from 'react';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { Button, HorizontalGroup, Modal, useStyles2 } from '@grafana/ui';
+import { FormProvider, useForm } from 'react-hook-form';
+
+import { ApiSchemas } from 'network/oncall-api/api.types';
+import { useCurrentIntegration } from 'pages/integration/OutgoingTab/OutgoingTab.hooks';
+import { useStore } from 'state/useStore';
+import { OmitReadonlyMembers } from 'utils/types';
+import { openNotification } from 'utils/utils';
+
+import { getCommonServiceNowConfigStyles } from './ServiceNow.styles';
+import { ServiceNowStatusSection } from './ServiceNowStatusSection';
+import { ServiceNowTokenSection } from './ServiceNowTokenSection';
+
+interface CompleteServiceNowConfigModalProps {
+  onHide: () => void;
+}
+
+interface FormFields {
+  additional_settings: ApiSchemas['AlertReceiveChannel']['additional_settings'];
+}
+
+export const CompleteServiceNowModal: React.FC<CompleteServiceNowConfigModalProps> = ({ onHide }) => {
+  const { alertReceiveChannelStore } = useStore();
+  const integration = useCurrentIntegration();
+
+  const formMethods = useForm<FormFields>({
+    values: {
+      additional_settings: {
+        ...integration.additional_settings,
+      },
+    },
+  });
+
+  const [isFormActionsDisabled, setIsFormActionsDisabled] = useState(false);
+
+  const styles = useStyles2(getStyles);
+  const { handleSubmit } = formMethods;
+
+  const { id } = integration;
+
+  return (
+    <Modal closeOnEscape={false} isOpen title={'Complete ServiceNow configuration'} onDismiss={onDismiss}>
+      <FormProvider {...formMethods}>
+        <form onSubmit={handleSubmit(onFormSubmit)}>
+          <div className={styles.border}>
+            <ServiceNowStatusSection />
+          </div>
+
+          <div className={styles.border}>
+            <ServiceNowTokenSection />
+          </div>
+
+          <div>
+            <HorizontalGroup justify="flex-end">
+              <Button variant="secondary" onClick={onFormAcknowledge} disabled={isFormActionsDisabled}>
+                Close
+              </Button>
+              <Button variant="primary" type="submit" disabled={isFormActionsDisabled}>
+                Proceed
+              </Button>
+            </HorizontalGroup>
+          </div>
+        </form>
+      </FormProvider>
+    </Modal>
+  );
+
+  async function onDismiss() {
```

Comment:
Can we remove this function and use `onDismiss={onFormAcknowledge` instead? Both x btn and Close should do exactly same thing so they can both use the same handler I suppose?